### PR TITLE
sql: add open txns gauge

### LINKED ISSUE #52005: metrics: Expose active sessions metric
It would be useful to expose a new metric in the Prometheus endpoint that tracks the number of currently active SQL sessions for troubleshooting and alerting purposes.

### LINKED ISSUE #57475: sql: gauge that tracks currently open SQL transactions
There is currently no way to see the number of open SQL transactions across the cluster as a timeseries metric. This is an important metric to add.

It should be called `sql.txns.open`.

### PR DESCRIPTION
**Pull request by jordanlewis**
This is a new gauge in `sql.txns.open` that tracks the number of open
transactions.

Closes #57475
Closes #52005

cc @a-entin @awoods187 

The dashboard now contains the following 3 graphs first:

1. Open SQL Sessions
2. Open SQL Transactions
3. Active SQL Queries

Release note (ui change): add Open SQL Transactions to SQL dashboard in
DB Console. Rename SQL Queries to SQL Statements in SQL metrics
dashboard. Remove references to "Distributed SQL Queries" when metrics
really are discussing all SQL queries.

### CODE REVIEW

**Review by yuzefovich — APPROVED** [review]
This is not exposed in the UI right now, right? Do we want a dashboard for it? If not, how is it accessible for users/SREs?

The code :lgtm: though.

> Reviewed 4 of 4 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/57477#-:-MNdmTl87IEfL1Wd1v2q:bd1jfhg)* status: :shipit: complete! 1 of 0 LGTMs obtained

**Comment by awoods187** [comment]
We definitely should add this to the DB console too.

**Comment by jordanlewis** [comment]
Added to the DB Console. I also took the liberty of homogenizing the DB Console's metrics in this department for clarity.

The top 3 metrics are now:

1. Open SQL Connections (renamed from SQL Connections)
2. Open SQL Transactions (the new metric)
3. Active SQL Queries (renamed from Active Distributed SQL Queries)

image

**Review by yuzefovich — APPROVED** [review]
> Reviewed 4 of 4 files at r2, 1 of 1 files at r3.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/57477)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale)

File: pkg/ui/src/views/cluster/containers/nodeGraphs/dashboards/sql.tsx [hunk 1]
```diff
@@ -40,6 +40,26 @@ export default function (props: GraphDashboardProps) {
...
+      </Axis>
+    </LineGraph>,
+
+    <LineGraph
+      title="Active SQL Queries"
+      sources={nodeSources}
+      tooltip={`The total number of SQL queries currently running ${tooltipSelection}.`}
+    >
+      <Axis label="queries">
+        <Metric name="cr.node.sql.distsql.queries.active" title="Active Queries" />
```
Comment (rafiss) [inline pkg/ui/src/views/cluster/containers/nodeGraphs/dashboards/sql.tsx:59]: drive-by comment -- does this metric capture all queries (not just distSQL)? is that new or the same?

should the comment on the `sql.distsql.queries.active` metric be updated in execinfra/metrics.go to say this?

File: pkg/ui/src/views/cluster/containers/nodeGraphs/dashboards/sql.tsx [hunk 2]
```diff
@@ -40,6 +40,26 @@ export default function (props: GraphDashboardProps) {
       </Axis>
     </LineGraph>,
 
+    <LineGraph
+      title="Open SQL Transactions"
+      sources={nodeSources}
+      tooltip={`The total number of SQL transactions currently open ${tooltipSelection}.`}
+    >
+      <Axis label="transactions">
+        <Metric name="cr.node.sql.txns.open" title="Open Transactions" />
```
Comment (rafiss) [inline pkg/ui/src/views/cluster/containers/nodeGraphs/dashboards/sql.tsx:49]: drive-by comment -- i'm curious if this includes open implicit transactions?

more fundamentally, i'm interested in how would i answer the question: "how many of my SQL connections are currently doing something?" which metric would i use for that? (no prob if the answer is none of these, i'm just trying to understand these metrics, not suggesting any changes)

**Comment by rafiss** [comment]
is there already a process for getting the new DB console chart documented here: https://www.cockroachlabs.com/docs/v20.2/ui-sql-dashboard.html (curious if this should be in the Release Note?)

**Comment by a-entin** [comment]
@jordanlewis and @awoods187 : one though is we use both terms - "connections" and "sessions" in the UI. While they are synonymous, it maybe better to pick one for consistency. I don't have a strong opinion about it since say ODBC/JDBC vernacular is connections and in our database users manage sessions. Yet it maybe worthwhile to define and perhaps even put in the doc a nice taxonomy, then revisit all user visible labels everywhere.

File: pkg/ui/src/views/cluster/containers/nodeGraphs/dashboards/sql.tsx (hunk 1 above)
Comment (awoods187) [reply pkg/ui/src/views/cluster/containers/nodeGraphs/dashboards/sql.tsx:59]: Yeah im curious on both of these points too

**Comment by awoods187** [comment]
> While they are synonymous, it may be better to pick one for consistency. 
I agree. I like the idea of naming this metric `open sessions` to match the sessions page. We can revisit the decision on sessions vs. connections in separate work.

For similar reasons, I think the last chart should be renamed from `queries` to `statements`.

I also like the hierarchy of sessions->transactions->statements as it matches the left-hand nav on those individual pages..

**Comment by a-entin** [comment]
I second Andy's suggestion.

A nit... isn't "open" implied in "open sessions"? What other sessions could there be. Can be omitted. Yet a clarification re user (or "client") sessions / transactions may be helpful to reflect if can be 1 adjective added to a label.  This is merely a thought, should in no way delay getting the essential new metric into the UI.

**Comment by awoods187** [comment]
I think we should backport this to 20.2.X as well.

File: pkg/ui/src/views/cluster/containers/nodeGraphs/dashboards/sql.tsx (hunk 1 above)
Comment (jordanlewis) [reply pkg/ui/src/views/cluster/containers/nodeGraphs/dashboards/sql.tsx:59]: All queries, yes

File: pkg/ui/src/views/cluster/containers/nodeGraphs/dashboards/sql.tsx (hunk 2 above)
Comment (jordanlewis) [reply pkg/ui/src/views/cluster/containers/nodeGraphs/dashboards/sql.tsx:49]: It does include implicit txns

Comment (jordanlewis) [reply pkg/ui/src/views/cluster/containers/nodeGraphs/dashboards/sql.tsx:49]: I think "doing something" would be satisfied by open txns, which cost resources even if the connection is idle. Maybe active queries would be slightly better, since those are doing "more work".

**Comment by jordanlewis** [comment]
Open isn't implied. A metric could count the cumulative number of sessions that have been opened ever.

**Review by dhartunian — APPROVED** [review]
:lgtm: 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/57477#-:-MQnn2TTA3XzkOoPD5QY:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (and 1 stale) (waiting on @rafiss and @yuzefovich)
